src/dataset/mod.rs

```diff
@@ -43,10 +50,113 @@ pub trait Float:
     + for<'a> SubAssign<&'a Self>
     + for<'a> DivAssign<&'a Self>
     + SampleUniform
+    + ScalarOperand
```

Comment:
Maybe we could add `num_traits::MulAdd<Output = F>` (see https://github.com/vbarrielle/sprs/issues/278#issuecomment-821520964) to ease the upgrade to `sprs 0.10`?

Reply:
yes! Haven't seen the post